File: php/1st_week/AtributePrivate/Buku.php
<?php
class Buku{
    
    //properties awal diset string kosong
    private $judul = "";
    private $pengarang = "";

    //deklarasi properties dengan constructor
    function __construct($j="", $p="")
    {
        $this->judul = $j;
        $this->pengarang = $p;
    }

    //function 
    function setJudul($j=""){
        $this->judul = $j;
    }

    function setPengarang($p=""){
        $this->pengarang = $p;
    }

    function getJudul(){
        return $this->judul;
    }

    function getPengarang(){
        return $this->pengarang;
    }
}
?>

File: cpp/1st_Week/AtributPrivate/Buku.cpp
#include <iostream>
#include <string>

using namespace std;

class Buku
{
    private:
        string judul;
        string pengarang;
    public:
    //constructor kosong
    Buku(){
    };
        
    //constructor dengan properties
    Buku(string j, string p){
        judul = j;
        pengarang = p;
    }

    //fungsi setter/mengganti judul dengan masukan baru
    void setJudul(string j){
        judul = j;
    }

    void setPengarang(string p){
        pengarang = p;
    }

    //fungsi getter/mengakses data di properties
    string getJudul(){
        return judul;
    }

    string getPengarang(){
        return pengarang;
    }

    //destructor/free
    ~Buku(){
    };
};
